python/lsst/daf/butler/registry/interfaces/_collections.py

```diff
@@ -207,6 +207,9 @@ def refresh(self, manager: CollectionManager) -> None:
             instances that may appear as its children.
         """
         self._children = self._load(manager)
+        # Update the reverse mapping (from child to parents) in the manager.
+        for child in self._children:
+            manager._parents_by_child[manager.find(child).key].add(self.key)
```

Comment:
What happens if a child is removed from a chain, should this code remove all previously known children of this chain from reverse mapping first?

Reply:
Good catch, fixed.